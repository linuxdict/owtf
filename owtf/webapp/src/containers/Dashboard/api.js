import Request from "../../utils/request";
import { API_BASE_URL } from "../../utils/constants";

function getHeaders() {
  return {
    headers: {
      "Content-Type": "application/x-www-form-urlencoded;charset=UTF-8",
      Authorization: `Bearer ${localStorage.getItem("token")}`
    }
  };
}

export function getErrorsAPI() {
  const requestURL = `${API_BASE_URL}errors/`;
  // Call our request helper (see 'utils/request')
  const options = getHeaders();
  const request = new Request(requestURL, options);
  return request.get.bind(request);
}

export function postErrorAPI() {
  const requestURL = `${API_BASE_URL}errors/`;
  const options = getHeaders();
  const request = new Request(requestURL, options);
  return request.post.bind(request);
}

export function deleteErrorAPI(action) {
  const error_id = action.error_id.toString();
  const requestURL = `${API_BASE_URL}errors/${error_id}/`;
  const options = getHeaders();
  const req_options = {
    responseAs: "text",
    ...options
  };
  const request = new Request(requestURL, req_options);
  return request.delete.bind(request);
}

export function getSeverityAPI() {
  const requestURL = `${API_BASE_URL}dashboard/severitypanel/`;
  // Call our request helper (see 'utils/request')
  const options = getHeaders();
  const request = new Request(requestURL, options);
  return request.get.bind(request);
}

export function getTargetSeverityAPI() {
  const requestURL = `${API_BASE_URL}targets/severitychart/`;
  // Call our request helper (see 'utils/request')
  const options = getHeaders();
  const request = new Request(requestURL, options);
  return request.get.bind(request);
}
