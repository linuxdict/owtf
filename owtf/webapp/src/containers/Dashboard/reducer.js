/*
 * ErrorReducer
 *
 * The reducer takes care of our data. Using actions, we can change our
 * application state.
 * To add a new action, add it to the switch statement in the reducer function
 *
 * Example:
 * case YOUR_ACTION_CONSTANT:
 *   return state.set('yourStateVariable', true);
 */

import { fromJS } from "immutable";
import { combineReducers } from "redux-immutable"; // combineReducers of 'redux' doesn't work with immutable.js

import {
  LOAD_ERRORS,
  LOAD_ERRORS_SUCCESS,
  LOAD_ERRORS_ERROR,
  CREATE_ERROR,
  CREATE_ERROR_SUCCESS,
  CREATE_ERROR_ERROR,
  DELETE_ERROR,
  DELETE_ERROR_SUCCESS,
  DELETE_ERROR_ERROR,
  LOAD_SEVERITY,
  LOAD_SEVERITY_SUCCESS,
  LOAD_SEVERITY_ERROR,
  LOAD_TARGET_SEVERITY,
  LOAD_TARGET_SEVERITY_SUCCESS,
  LOAD_TARGET_SEVERITY_ERROR,
} from "./constants";

// The initial state of the errors.
const initialErrorsState = fromJS({
  loading: true,
  error: false,
  errors: false
});

export function errorsLoadReducer(state = initialErrorsState, action) {
  switch (action.type) {
    case LOAD_ERRORS:
      return state
        .set("loading", true)
        .set("error", false)
        .set("errors", false);
    case LOAD_ERRORS_SUCCESS:
      return state
        .set("loading", false)
        .set("error", false)
        .set("errors", action.errors);
    case LOAD_ERRORS_ERROR:
      return state
        .set("loading", false)
        .set("error", action.error)
        .set("errors", false);
    default:
      return state;
  }
}

// The initial state of the error create
const initialCreateState = fromJS({
  loading: false,
  error: false
});

export function errorCreateReducer(state = initialCreateState, action) {
  switch (action.type) {
    case CREATE_ERROR:
      return state.set("loading", true).set("error", false);
    case CREATE_ERROR_SUCCESS:
      return state.set("loading", false).set("error", false);
    case CREATE_ERROR_ERROR:
      return state.set("error", action.error).set("loading", false);
    default:
      return state;
  }
}

// The initial state of the error delete
const initialDeleteState = fromJS({
  loading: false,
  error: false
});

export function errorDeleteReducer(state = initialDeleteState, action) {
  switch (action.type) {
    case DELETE_ERROR:
      return state.set("loading", true).set("error", false);
    case DELETE_ERROR_SUCCESS:
      return state.set("loading", false).set("error", false);
    case DELETE_ERROR_ERROR:
      return state.set("error", action.error).set("loading", false);
    default:
      return state;
  }
}

// The initial state of the severity.
const initialSeverityState = fromJS({
  loading: true,
  error: false,
  severity: false
});

export function severityLoadReducer(state = initialSeverityState, action) {
  switch (action.type) {
    case LOAD_SEVERITY:
      return state
        .set("loading", true)
        .set("error", false)
        .set("severity", false);
    case LOAD_SEVERITY_SUCCESS:
      return state
        .set("loading", false)
        .set("error", false)
        .set("severity", action.severity);
    case LOAD_SEVERITY_ERROR:
      return state
        .set("loading", false)
        .set("error", action.error)
        .set("severity", false);
    default:
      return state;
  }
}

// The initial state of the last target severity.
const initialTargetSeverityState = fromJS({
  loading: true,
  error: false,
  targetSeverity: false
});

export function targetSeverityLoadReducer(state = initialTargetSeverityState, action) {
  switch (action.type) {
    case LOAD_TARGET_SEVERITY:
      return state
        .set("loading", true)
        .set("error", false)
        .set("targetSeverity", false);
    case LOAD_TARGET_SEVERITY_SUCCESS:
      return state
        .set("loading", false)
        .set("error", false)
        .set("targetSeverity", action.targetSeverity);
    case LOAD_TARGET_SEVERITY_ERROR:
      return state
        .set("loading", false)
        .set("error", action.error)
        .set("targetSeverity", false);
    default:
      return state;
  }
}

export default combineReducers({
  load: errorsLoadReducer,
  create: errorCreateReducer,
  delete: errorDeleteReducer,
  loadSeverity: severityLoadReducer,
  loadTargetSeverity: targetSeverityLoadReducer,
});
