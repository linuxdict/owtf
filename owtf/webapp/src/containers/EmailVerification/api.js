import Request from "../../utils/request";
import { API_BASE_URL } from "../../utils/constants";

const options = {
  headers: {
    "Content-Type": "application/x-www-form-urlencoded;charset=UTF-8"
  }
};

export function confirmEmailGenerateAPI() {
  const requestURL = `${API_BASE_URL}generate/confirm_email/`;
  const request = new Request(requestURL, options);
  return request.post.bind(request);
}

export function verifyEmailGenerateAPI(link) {
  const requestURL = `${API_BASE_URL}verify/confirm_email/` + link;
  const request = new Request(requestURL, options);
  return request.get.bind(request);
}
