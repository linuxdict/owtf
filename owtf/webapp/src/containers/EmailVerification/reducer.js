/*
 * EmailVerificationReducer
 *
 * Changes the redux store state on emailSend
 */

import { fromJS } from "immutable";
import { combineReducers } from "redux-immutable"; // combineReducers of 'redux' doesn't work with immutable.js

import {
  EMAIL_SEND_START,
  EMAIL_SEND_SUCCESS,
  EMAIL_SEND_FAIL,
  EMAIL_VERIFICATION_START,
  EMAIL_VERIFICATION_SUCCESS,
  EMAIL_VERIFICATION_FAIL
} from "./constants";

// The initial state of the email send.
const initialEmailSendState = fromJS({
  error: false,
  loading: false
});

export function emailSendReducer(state = initialEmailSendState, action) {
  switch (action.type) {
    case EMAIL_SEND_START:
      return state.set("loading", true).set("error", false);
    case EMAIL_SEND_SUCCESS:
      return state.set("loading", false).set("error", false);
    case EMAIL_SEND_FAIL:
      return state.set("loading", false).set("error", action.error);
    default:
      return state;
  }
}

// The initial state of the email verify.
const initialEmailVerificationState = fromJS({
  error: false,
  loading: false
});

export function emailVerificationReducer(
  state = initialEmailVerificationState,
  action
) {
  switch (action.type) {
    case EMAIL_VERIFICATION_START:
      return state.set("loading", true).set("error", false);
    case EMAIL_VERIFICATION_SUCCESS:
      return state.set("loading", false).set("error", false);
    case EMAIL_VERIFICATION_FAIL:
      return state.set("loading", false).set("error", action.error);
    default:
      return state;
  }
}

export default combineReducers({
  send: emailSendReducer,
  verify: emailVerificationReducer
});
