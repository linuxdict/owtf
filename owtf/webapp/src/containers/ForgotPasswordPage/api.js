import Request from "../../utils/request";
import { API_BASE_URL } from "../../utils/constants";

export function emailAPI() {
  const requestURL = `${API_BASE_URL}generate/otp/`;
  const options = {
    headers: {
      "Content-Type": "application/x-www-form-urlencoded;charset=UTF-8"
    }
  };
  const request = new Request(requestURL, options);
  return request.post.bind(request);
}
