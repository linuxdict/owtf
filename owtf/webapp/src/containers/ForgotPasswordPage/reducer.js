import { fromJS } from "immutable";
import { combineReducers } from "redux-immutable"; // combineReducers of 'redux' doesn't work with immutable.js

import {
  FORGOT_PASSWORD_EMAIL_START,
  FORGOT_PASSWORD_EMAIL_SUCCESS,
  FORGOT_PASSWORD_EMAIL_FAIL
} from "./constants";

// The initial state of the forgot password
const initialForgotPasswordEmailState = fromJS({
  error: false,
  loading: false,
  emailOrUsername: null
});

export function forgotPasswordEmailReducer(
  state = initialForgotPasswordEmailState,
  action
) {
  switch (action.type) {
    case FORGOT_PASSWORD_EMAIL_START:
      return state
        .set("loading", true)
        .set("error", false)
        .set("emailOrUsername", null);
    case FORGOT_PASSWORD_EMAIL_SUCCESS:
      return state
        .set("loading", false)
        .set("error", false)
        .set("emailOrUsername", action.emailOrUsername);
    case FORGOT_PASSWORD_EMAIL_FAIL:
      return state
        .set("loading", false)
        .set("error", action.error)
        .set("emailOrUsername", null);
    default:
      return state;
  }
}

export default combineReducers({
  forgot: forgotPasswordEmailReducer
});
