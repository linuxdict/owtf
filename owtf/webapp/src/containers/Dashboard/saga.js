/**
 * Fetch, Create and Change the error from API
 */

import { call, put, takeLatest } from "redux-saga/effects";
import { LOAD_ERRORS, CREATE_ERROR, DELETE_ERROR, LOAD_SEVERITY, LOAD_TARGET_SEVERITY } from "./constants";
import {
  loadErrors,
  errorsLoaded,
  errorsLoadingError,
  errorCreated,
  errorCreatingError,
  errorDeleted,
  errorDeletingError,
  severityLoaded,
  severityLoadingError,
  targetSeverityLoaded,
  targetSeverityLoadingError,
} from "./actions";
import { getErrorsAPI, postErrorAPI, deleteErrorAPI, getSeverityAPI, getTargetSeverityAPI } from "./api";
import { Dashboard } from "./index";
import "@babel/polyfill";

const dashboard = new Dashboard();
/**
 * Fetch Errors request/response handler
 */
export function* getErrors() {
  const fetchAPI = getErrorsAPI();
  try {
    // Call our request helper (see 'utils/request')
    const errors = yield call(fetchAPI);
    yield put(errorsLoaded(errors));
  } catch (error) {
    yield put(errorsLoadingError(error));
  }
}

/**
 * Post Error request/response handler
 */
export function* postError(action) {
  const postAPI = postErrorAPI();
  try {
    yield call(postAPI, action.post_data);
    yield put(errorCreated());
    yield put(loadErrors());
  } catch (error) {
    yield put(errorCreatingError(error));
  }
}

/**
 * Delete Error request/response handler
 */
export function* deleteError(action) {
  const deleteAPI = deleteErrorAPI(action);
  try {
    yield call(deleteAPI);
    yield put(errorDeleted());
    yield put(loadErrors());
    dashboard.toasterSuccess(action.error_id, "deleteError");
  } catch (error) {
    yield put(errorDeletingError(error));
    dashboard.toasterError(error);
  }
}

/**
 * Fetch Severity request/response handler
 */
export function* getSeverity() {
  const fetchAPI = getSeverityAPI();
  try {
    // Call our request helper (see 'utils/request')
    const severity = yield call(fetchAPI);
    yield put(severityLoaded(severity.data));
  } catch (error) {
    yield put(severityLoadingError(error));
  }
}

/**
 * Fetch last target severity request/response handler
 */
export function* getTargetSeverity() {
  const fetchAPI = getTargetSeverityAPI();
  try {
    // Call our request helper (see 'utils/request')
    const targetSeverity = yield call(fetchAPI);
    yield put(targetSeverityLoaded(targetSeverity.data));
  } catch (error) {
    yield put(targetSeverityLoadingError(error));
  }
}

/**
 * Root saga manages watcher lifecycle
 */
export default function* errorsSaga() {
  // Watches for LOAD_ERROR actions and calls getError when one comes in.
  // By using `takeLatest` only the result of the latest API call is applied.
  // It returns task descriptor (just like fork) so we can continue execution
  // It will be cancelled automatically on component unmount
  yield takeLatest(LOAD_ERRORS, getErrors);
  yield takeLatest(CREATE_ERROR, postError);
  yield takeLatest(DELETE_ERROR, deleteError);
  yield takeLatest(LOAD_SEVERITY, getSeverity);
  yield takeLatest(LOAD_TARGET_SEVERITY, getTargetSeverity);
}
