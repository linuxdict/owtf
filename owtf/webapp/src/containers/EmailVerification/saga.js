import { call, put, takeLatest } from "redux-saga/effects";
import { confirmEmailGenerateAPI, verifyEmailGenerateAPI } from "./api";
import { EMAIL_SEND_START, EMAIL_VERIFICATION_START } from "./constants";
import {
  emailSendSuccess,
  emailSendFail,
  emailVerificationSuccess,
  emailVerificationFail
} from "./actions";
import { toaster } from "evergreen-ui";
import { push } from "react-router-redux";
import "@babel/polyfill";

/**
 * Send the email verification link to the email of the user using API
 */
export function* postEmailToGenerateAPI(action) {
  const emailgenerateAPI = confirmEmailGenerateAPI();
  try {
    const emailData = {
      email: action.email
    };
    const responseData = yield call(emailgenerateAPI, emailData);
    if (responseData.data["message"] === "Email send successful") {
      toaster.success(responseData.data["message"]);
      yield put(emailSendSuccess(responseData.data["message"]));
    }
  } catch (error) {
    toaster.danger("Email send unsuccessful");
    yield put(emailSendFail(error));
  }
}

/**
 * Verify the email verification link of the user using API
 */
export function* postToVerificationAPI(action) {
  const emailgenerateAPI = verifyEmailGenerateAPI(action.link);
  try {
    let pathname;
    const responseData = yield call(emailgenerateAPI);
    if (responseData.data["status"] === "success") {
      if (responseData.data["message"] === "Email Verified") {
        pathname = "/login";
      } else if (responseData.data["message"] === "Link Expired") {
        pathname = "/email-send/";
      } else if (responseData.data["message"] === "Invalid Link") {
        pathname = "/signup";
      }
      yield put(push(pathname));
      toaster.success(responseData.data["message"]);
      yield put(emailVerificationSuccess(responseData.data["message"]));
    }
  } catch (error) {
    toaster.danger("Email verification unsuccessful");
    yield put(emailVerificationFail(error));
  }
}

/**
 * Root saga manages watcher lifecycle
 */
export default function* emailVerificationSaga() {
  yield takeLatest(EMAIL_SEND_START, postEmailToGenerateAPI);
  yield takeLatest(EMAIL_VERIFICATION_START, postToVerificationAPI);
}
