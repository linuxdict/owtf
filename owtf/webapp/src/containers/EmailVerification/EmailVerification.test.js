import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import "../../setupTests";
import { findByTestAtrr, checkProps } from "../../utils/testUtils";
import { EmailSendPage } from "./index";
import { fromJS } from "immutable";

import {
  emailSendSuccess,
  emailSendFail,
  emailVerificationSuccess,
  emailVerificationFail
} from "./actions";

import {
  EMAIL_SEND_START,
  EMAIL_SEND_SUCCESS,
  EMAIL_SEND_FAIL,
  EMAIL_VERIFICATION_START,
  EMAIL_VERIFICATION_SUCCESS,
  EMAIL_VERIFICATION_FAIL
} from "./constants";

import { postEmailToGenerateAPI, postToVerificationAPI } from "./saga";
import { runSaga } from "redux-saga";
import * as api from "./api";

import { emailSendReducer, emailVerificationReducer } from "./reducer";

describe("EmailVerification component", () => {
  describe("Testing dumb email verification component", () => {
    let wrapper;
    let props;
    beforeEach(() => {
      props = {
        onResend: jest.fn(),
        email: "test_user@test.com"
      };
      wrapper = shallow(<EmailSendPage {...props} />);
    });
    it("Should have correct prop-types", () => {
      const expectedProps = {
        onResend: () => {},
        email: "test_user@test.com"
      };
      const propsErr = checkProps(EmailSendPage, expectedProps);
      expect(propsErr).toBeUndefined();
    });

    it("Should render without errors", () => {
      const component = findByTestAtrr(wrapper, "emailSendPageComponent");
      expect(component.length).toBe(1);
      expect(toJson(component)).toMatchSnapshot();
    });

    it("Should correctly render its sub-components", () => {
      const heading = wrapper.find("withTheme(Heading)");
      const paragraph = wrapper.find("withTheme(Paragraph)");
      const button = wrapper.find("withTheme(Button)");

      expect(heading.length).toBe(1);
      expect(heading.at(0).props().children).toEqual("Email Sent");
      expect(paragraph.length).toBe(4);
      expect(paragraph.at(0).props().children).toEqual(
        "We have sent a mail to verify your email address."
      );
      expect(paragraph.at(3).props().children).toEqual(
        "If you don't find it in your inbox, check spam folder."
      );
      expect(button.length).toBe(1);
      expect(button.at(0).props().children).toEqual("Resend");
    });

    it("Should call onResend on Resend button click", () => {
      expect(props.onResend.mock.calls.length).toBe(0);
      const resendButton = wrapper.find("withTheme(Button)");
      resendButton.simulate("click");
      expect(props.onResend.mock.calls.length).toBe(1);
    });
  });

  describe("Testing the sagas", () => {
    describe("Testing postEmailToGenerateAPI saga", () => {
      let fakeAction;
      beforeEach(() => {
        fakeAction = {
          type: EMAIL_SEND_START,
          email: "test_user@test.com"
        };
      });

      it("Should send an email in case of success", async () => {
        const successData = {
          status: "success",
          data: { status: "success", message: "Email send successful" }
        };

        const dispatchedActions = [];
        api.confirmEmailGenerateAPI = jest.fn(() =>
          jest.fn(() => Promise.resolve(successData))
        );

        const fakeStore = {
          getState: () => ({ state: "test" }),
          dispatch: action => dispatchedActions.push(action)
        };

        await runSaga(fakeStore, postEmailToGenerateAPI, fakeAction).done;
        expect(api.confirmEmailGenerateAPI.mock.calls.length).toBe(1);
        expect(dispatchedActions).toContainEqual(
          emailSendSuccess(successData["data"]["message"])
        );
      });

      it("Should handle send email error in case of failure", async () => {
        const dispatchedActions = [];
        const error = "API server is down";

        api.confirmEmailGenerateAPI = jest.fn(() =>
          jest.fn(() => Promise.reject(error))
        );

        const fakeStore = {
          getState: () => ({ state: "test" }),
          dispatch: action => dispatchedActions.push(action)
        };

        await runSaga(fakeStore, postEmailToGenerateAPI, fakeAction).done;
        expect(api.confirmEmailGenerateAPI.mock.calls.length).toBe(1);
        expect(dispatchedActions).toContainEqual(emailSendFail(error));
      });
    });

    describe("Testing postToVerificationAPI saga", () => {
      let fakeAction;
      beforeEach(() => {
        fakeAction = {
          type: EMAIL_VERIFICATION_START,
          link: "test_link"
        };
      });

      it("Should verify a link in case of success", async () => {
        const successData = {
          status: "success",
          data: { status: "success", message: "Email Verified" }
        };

        const dispatchedActions = [];
        api.verifyEmailGenerateAPI = jest.fn(() =>
          jest.fn(() => Promise.resolve(successData))
        );

        const fakeStore = {
          getState: () => ({ state: "test" }),
          dispatch: action => dispatchedActions.push(action)
        };

        await runSaga(fakeStore, postToVerificationAPI, fakeAction).done;
        expect(api.verifyEmailGenerateAPI.mock.calls.length).toBe(1);
        expect(dispatchedActions).toContainEqual(
          emailVerificationSuccess(successData["data"]["message"])
        );
      });

      it("Should handle link verification error in case of failure", async () => {
        const dispatchedActions = [];
        const error = "API server is down";
        api.verifyEmailGenerateAPI = jest.fn(() =>
          jest.fn(() => Promise.reject(error))
        );

        const fakeStore = {
          getState: () => ({ state: "test" }),
          dispatch: action => dispatchedActions.push(action)
        };

        await runSaga(fakeStore, postToVerificationAPI, fakeAction).done;
        expect(api.verifyEmailGenerateAPI.mock.calls.length).toBe(1);
        expect(dispatchedActions).toContainEqual(emailVerificationFail(error));
      });
    });
  });

  describe("Testing reducers", () => {
    describe("Testing emailSendReducer", () => {
      let initialState;
      beforeEach(() => {
        initialState = {
          error: false,
          loading: false
        };
      });

      it("Should return the initial state", () => {
        const newState = emailSendReducer(undefined, {});
        expect(newState.toJS()).toEqual(initialState);
      });

      it("Should handle EMAIL_SEND_START", () => {
        const newState = emailSendReducer(undefined, {
          type: EMAIL_SEND_START
        });

        expect(newState.toJS()).toEqual({
          loading: true,
          error: false
        });
      });

      it("Should handle EMAIL_SEND_SUCCESS", () => {
        const newState = emailSendReducer(
          fromJS({
            loading: true,
            error: true
          }),
          {
            type: EMAIL_SEND_SUCCESS
          }
        );

        expect(newState.toJS()).toEqual({
          loading: false,
          error: false
        });
      });

      it("Should handle EMAIL_SEND_FAIL", () => {
        const err = "Test sessions loading error";
        const newState = emailSendReducer(
          fromJS({
            loading: true,
            error: true
          }),
          {
            type: EMAIL_SEND_FAIL,
            error: err
          }
        );

        expect(newState.toJS()).toEqual({
          loading: false,
          error: err
        });
      });
    });
    describe("Testing emailVerificationReducer", () => {
      let initialState;
      beforeEach(() => {
        initialState = {
          error: false,
          loading: false
        };
      });

      it("Should return the initial state", () => {
        const newState = emailVerificationReducer(undefined, {});
        expect(newState.toJS()).toEqual(initialState);
      });

      it("Should handle EMAIL_VERIFICATION_START", () => {
        const link = "test_user_link";
        const newState = emailVerificationReducer(undefined, {
          type: EMAIL_VERIFICATION_START,
          link: link
        });

        expect(newState.toJS()).toEqual({
          loading: true,
          error: false
        });
      });

      it("Should handle EMAIL_VERIFICATION_SUCCESS", () => {
        const msg = "Email Verified";
        const newState = emailVerificationReducer(
          fromJS({
            loading: true,
            error: true
          }),
          {
            type: EMAIL_VERIFICATION_SUCCESS,
            msg: msg
          }
        );

        expect(newState.toJS()).toEqual({
          loading: false,
          error: false
        });
      });

      it("Should handle EMAIL_VERIFICATION_FAIL", () => {
        const err = "Test sessions loading error";
        const newState = emailVerificationReducer(
          fromJS({
            loading: true,
            error: true
          }),
          {
            type: EMAIL_VERIFICATION_FAIL,
            error: err
          }
        );

        expect(newState.toJS()).toEqual({
          loading: false,
          error: err
        });
      });
    });
  });
});
