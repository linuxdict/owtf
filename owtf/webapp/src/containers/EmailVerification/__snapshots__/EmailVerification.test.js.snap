// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmailVerification component Testing dumb email verification component Should render without errors 1`] = `
<withTheme(Pane)
  data-test="emailSendPageComponent"
  marginY={100}
>
  <withTheme(Pane)
    elevation={1}
    justifyContent="center"
    margin="auto"
    padding={5}
    width="35%"
  >
    <withTheme(Heading)
      marginBottom={20}
      paddingTop={20}
      size={700}
      textAlign="center"
    >
      Email Sent
    </withTheme(Heading)>
    <withTheme(Paragraph)
      marginBottom={5}
      marginLeft="20%"
      size={300}
      width="60%"
    >
      We have sent a mail to verify your email address.
    </withTheme(Paragraph)>
    <withTheme(Paragraph)
      marginLeft="20%"
      size={300}
      width="60%"
    >
      If you don't receive, please click here to:
      <br />
      <withTheme(Link)>
        <withTheme(Button)
          appearance="primary"
          intent="none"
          justifyContent="center"
          marginBottom={10}
          marginLeft="40%"
          marginTop={10}
          onClick={[Function]}
          width="20%"
        >
          Resend
        </withTheme(Button)>
      </withTheme(Link)>
    </withTheme(Paragraph)>
    <withTheme(Paragraph)
      marginBottom={15}
      marginLeft="20%"
      size={300}
      width="60%"
    >
      Once you verify, 
      <withTheme(Link)
        href="/login"
      >
        Login
      </withTheme(Link)>
       here
    </withTheme(Paragraph)>
    <withTheme(Paragraph)
      marginBottom={20}
      marginLeft="20%"
      size={300}
      width="60%"
    >
      If you don't find it in your inbox, check spam folder.
    </withTheme(Paragraph)>
  </withTheme(Pane)>
</withTheme(Pane)>
`;
